api/app/services/video_thumbnail/fetcher.rb

```diff
@@ -0,0 +1,32 @@
+module VideoThumbnail
+  # This class is responsible for fetching tweets related to a given project
+  class Fetcher
+    def fetch(video)
+      case video.external_type
```

Comment:
You're coupling these models too closely by relying on the value of external_type here. What if we change how we store this data? Instead, let the resource have the final say as to what it is.
```
def fetch(resource)
  fetch_youtube_thumbnail if resource.is_youtube?
  fetch_vimdeo_thumbnail  if resource.is_vimeo?
end
```